proxy/proxy.go

```diff
@@ -239,6 +240,29 @@ func (s *Server) Stop() {
 	s.wg.Wait()
 }
 
+type dirWithoutIndex struct {
+	fs http.FileSystem
+}
+
+func (d dirWithoutIndex) Open(name string) (http.File, error) {
+	f, err := d.fs.Open(name)
+
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
There's one exception we need to make here: `/`

If the name is `/`, you can return the file descriptor normally.  Otherwise, the UI will not work.  😄 

Making this change causes the currently failing test (`systemtestSuite.TestUIResponseHeaders`) to pass, so we'll need to add another one which hits a different directory like `/assets` or something.